### PR TITLE
Add null pointer, let free(null) be nop

### PR DESCRIPTION
**Pull request by aqjune**
This PR adds null pointer support and makes free(null) be noop.

### CODE REVIEW

File: ir/value.h [hunk 1]
```diff
@@ -74,6 +74,14 @@ class VoidValue final : public Value {
 };
 
 
+class NullPointerValue final : public Value {
```
Comment (nunoplopes) [inline ir/value.h]: Why do we need a NullPointerValue and a NullPointerConst?
Isn't the const sufficient?

File: ir/memory.cpp [hunk 2]
```diff
@@ -254,9 +265,11 @@ expr Memory::alloc(const expr &bytes, unsigned align, bool heap) {
 
 void Memory::free(const expr &ptr) {
   Pointer p(*this, ptr);
-  state->addUB(p.get_offset() == 0);
-  state->addUB(p.is_block_alive());
-  state->addUB(p.is_at_heap());
+  auto isNullPointer = p.get_bid() == 0 && p.get_offset() == 0;
```
Comment (nunoplopes) [inline ir/memory.cpp]: maybe create a method in Pointer, since nullness checks will be frequent I guess.

File: ir/value.h (hunk 1 above)
Comment (aqjune) [reply ir/value.h]: Yeah, NullPointerValue was enough. I removed NullPointerConst

File: ir/memory.cpp (hunk 2 above)
Comment (aqjune) [reply ir/memory.cpp]: Done

**Comment by aqjune** [comment]
I applied the comments. :)
Running llvm's unit tests shows +12 failures:
```
Testing Time: 1163.97s -> 1186.59s
Expected Passes    : 4494 -> 4482
Expected Failures  : 25
Unsupported Tests  : 320 
Unexpected Failures: 121 -> 133
```
Here is the list of files:
```
Transforms/SimplifyCFG/PhiEliminate3.ll
Transforms/NewGVN/rle-nonlocal.ll
Transforms/NewGVN/2008-12-14-rle-reanalyze.ll
Transforms/LICM/2007-09-17-PromoteValue.ll
Transforms/JumpThreading/phi-known.ll
Transforms/InstCombine/gep-vector.ll
Transforms/GVN/equality-assume.ll
Transforms/GVN/non-integral-pointers.ll
Transforms/GVN/rle-nonlocal.ll
Transforms/DeadStoreElimination/pr11390.ll
Transforms/CorrelatedValuePropagation/sub.ll
Transforms/CorrelatedValuePropagation/basic.ll
```
I'll investigate why and make a separate PR.

**Pull request merged**